### ISSUE TITLE
[Bug] deny-egress netpol allows all traffic instead

### ISSUE THREAD

**Issue by synack-security** [kubecost/cost-analyzer-helm-chart#2816]
The deny-egress netpol actually allows all egress.
`  egress:
  - to:
    - namespaceSelector: {}` 
The above should allow all traffic to all namespaces.  [Here](https://github.com/ahmetb/kubernetes-network-policy-recipes/blob/master/05-allow-traffic-from-all-namespaces.md) is an example of ingress policy that allows all traffic similar to the egress above.

https://github.com/kubecost/cost-analyzer-helm-chart/blob/705bfc95f6b9cd0ed412fc2f2a8369ba292cc853/cost-analyzer/templates/cost-analyzer-network-policy.yaml#L19

A valid netpol that denies all egress traffic which you can find here https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/service/networking/network-policy-default-deny-all.yaml:

```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: deny-egress
spec:
  podSelector: {}
  policyTypes:
  - Egress
```

**Comment by AjayTripathy** [comment]
Thanks for this report. This should be a quick fix. @jessegoodier can you help triage?

**Comment by chipzoller** [comment]
Sent fix in #2821
